bindings/cpp/array.h

```diff
@@ -0,0 +1,536 @@
+// copyright, Alex Meakins and Qingfeng Xia  @ UKAEA
```

Comment:
This seems like overkill for what we need. I also think it shouldn't be coupled to NumPy.  My original suggestion was for a short Array<T> class which essentially just does the flattening/unflattening and stores the data. You can then populate it from numpy from outside the class.

Reply:
it is also open source code, compatible with this repo,  and it is well tested. it support all numpy dtype